type-definitions/Immutable.d.ts

```diff
@@ -2431,7 +2431,8 @@ declare module Immutable {
      * notSetValue will be returned if provided. Note that this scenario would
      * produce an error when using Flow or TypeScript.
      */
-    get<K extends keyof TProps>(key: K, notSetValue: any): TProps[K];
+    get<K extends keyof TProps>(key: K, notSetValue?: any): TProps[K];
+    get<T>(key: string, notSetValue: T): T;
```

Comment:
What if the key *is* set and the type of `TProps[key]` is different from the `T` value? Couldn't this result in a program that type checks but is incorrect?

Reply:
Do Records allow setting keys that are not in `TProps`?